tests/HttpClient/HttpPluginClientBuilderTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * (c) Packagist Conductors GmbH <contact@packagist.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PrivatePackagist\ApiClient\HttpClient\Plugin;
+
+use GuzzleHttp\Psr7\HttpFactory;
+use GuzzleHttp\Psr7\Response;
+use Http\Client\Common\HttpMethodsClientInterface;
+use Http\Message\MessageFactory\GuzzleMessageFactory;
+use Http\Message\RequestMatcher as RequestMatcherInterface;
+use Http\Mock\Client as MockClient;
+use PHPUnit\Framework\TestCase;
+use PrivatePackagist\ApiClient\HttpClient\HttpPluginClientBuilder;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class HttpPluginClientBuilderTest extends TestCase
+{
+    /** @dataProvider provideRequestFactories */
+    public function testRequestFactory(?object $factory, ?string $expectedException): void
```

Comment:
Can this be two different tests? One that throws the exception and one that doesn't? Makes it easier to see what is actually necessary for the test case with the exception